### PR TITLE
make webpack path strings os-independent

### LINKED ISSUE #6192: Current webpack config doesn't recognise windows paths
### Describe the bug
Bit's current webpack config contains path strings which are incompatible with windows paths.
e.g. https://github.com/teambit/bit/blob/77a1fa577e74fabf751ce1d58ee981986d9d9b4a/scopes/react/react-native/webpack/webpack-transformers.ts#L7

### Steps to Reproduce
1. On a windows machine, clone the [react-native demo repo](https://github.com/bit-demos/react-native) and follow the instructions to run the local dev server. You will get an error which classically means that react-native-* packages are not being transpiled to make them compatible with a web environment.
1. Do the same on a *nix machine and there's no error and everything runs well

### Expected Behavior
The webpack paths should be os-independent, e.g. using path.join to add the os-specific separator at build time

### Screenshots, exceptions and logs
If applicable, add screenshots, exceptions, and logs to help explain your problem.

### Specifications
* Bit version: 0.0.799
* Workspace type: (harmony)
* Node version: 16.15.1
* npm / yarn version:
* Platform: Windows

_for harmony workspace_

* relevant env - react-native
* relevant aspects

### PR DESCRIPTION
**Pull request by benjgil**
## Proposed Changes

- make webpack path strings os-resilient/independent
- fix #6192 
-

### CODE REVIEW

File: scopes/react/react-native/webpack/webpack-transformers.ts [hunk 1]
```diff
@@ -1,10 +1,11 @@
+import { join } from 'path';
 import { WebpackConfigTransformer, WebpackConfigMutator, WebpackConfigTransformContext } from '@teambit/webpack';
 import { get, set } from 'lodash';
 import { reactNativeAlias } from './react-native-alias';
 
 const reactNativePackagesRule = {
   test: /\.(jsx?|tsx?)$/,
-  include: [/node_modules\/react-native-/],
+  include: [new RegExp(join('node_modules$', 'react-native-/'))],
```
Comment (KutnerUri) [inline scopes/react/react-native/webpack/webpack-transformers.ts:8]: this might create wrong regex. maybe just replace `\/` with `[\\\/]`?

**Review by KutnerUri — APPROVED** [review]

File: scopes/react/react-native/webpack/webpack-transformers.ts (hunk 1 above)
Comment (GiladShoham) [reply scopes/react/react-native/webpack/webpack-transformers.ts]: I don't think the new RegExp is really needed.
also, maybe better to use this component - https://github.com/teambit/bit/blob/d7c9c7e36b7b8f9bd91ea4e80d991804d9ff3fa9/scopes/toolbox/path/to-windows-compatible-path/to-windows-compatible-path.ts#L7

**Review by KutnerUri — APPROVED** [review]

File: scopes/react/react/webpack/webpack.config.base.ts [hunk 2]
```diff
@@ -70,13 +71,13 @@ export default function (isEnvProduction = false): Configuration {
       extensions: moduleFileExtensions.map((ext) => `.${ext}`),
 
       alias: {
-        'react/jsx-dev-runtime': require.resolve('react/jsx-dev-runtime.js'),
-        'react/jsx-runtime': require.resolve('react/jsx-runtime.js'),
-        'react-dom/server': require.resolve('react-dom/server'),
+        [join('react', 'jsx-dev-runtime')]: require.resolve('react/jsx-dev-runtime.js'),
+        [join('react', 'jsx-runtime')]: require.resolve('react/jsx-runtime.js'),
+        [join('react-dom', 'server')]: require.resolve('react-dom/server'),
```
Comment (GiladShoham) [inline scopes/react/react/webpack/webpack.config.base.ts:76]: is this really required? i'm pretty sure webpack should know to handle this

Comment (benjgil) [reply scopes/react/react/webpack/webpack.config.base.ts:76]: we've seen a lot of issues with jsx-runtime not being loaded recently. Maybe it's just for windows users and webpack doesnt deal with this well?

File: scopes/react/react-native/webpack/webpack-transformers.ts (hunk 1 above)
Comment (benjgil) [reply scopes/react/react-native/webpack/webpack-transformers.ts]: that component wont help - it doesnt change / to\\, just a single to a double
the problem here is that the / (or \/ as escaped /) isnt being matched to the windows path

Comment (benjgil) [reply scopes/react/react-native/webpack/webpack-transformers.ts]: What would you do instead of RegExp? path.join wont produce a regular expression

**Review by GiladShoham — APPROVED** [review]

**Pull request merged**